mindmeld/components/question_answerer.py

```diff
@@ -451,803 +528,2178 @@ def _generate_mapping_data(embedder_model, embedding_fields):
+        and list of strings. Any other data type (eg. dictionary) is currently not supported and
+        is marked as an `unknown` data type. Such unknown data types fields do not have any
+        associated resolvers.
+        """
+
+        DATE_FORMATS = ('%Y', '%d %b', '%d %B', '%b %d, %Y', '%b %d, %Y', '%B %d, %Y',
+                        '%B %d %Y', '%m/%d/%Y', '%m/%d/%y', '%b %Y', '%B %Y', '%b %d,%Y')
+
+        def __init__(self, index_name, field_name):
+
+            # details to establish a scoped field name
+            self.index_name = index_name
+            self.field_name = field_name
+
+            # vars that contain data of the field
+            self.data_type = None
+            self.id2value = {}  # warning: duplicate data also exist in resolver object if created
+            self.hash = None  # to identify any data changes before build resolver
+
+            # details to set required resolvers
+            self.processor_type = None
+            self.has_text_resolver = None
+            self.has_embedding_resolver = None
+
+            # required resolvers
+            self._text_resolver = None  # an entity resolver if string type data
+            self._embedding_resolver = None  # an embedding based entity resolver
+
+        def __repr__(self):
+            return f"{self.__class__.__name__} " \
+                   f"field_name: {self.field_name} " \
+                   f"data_type: {self.data_type} " \
+                   f"has_text_resolver: {self.has_text_resolver} " \
+                   f"has_embedding_resolver: {self.has_embedding_resolver}"
+
+        @classmethod
+        def from_metadata(cls, cache_object: Bunch):
+            field_resource = cls(index_name=cache_object.index_name,
+                                 field_name=cache_object.field_name)
+            field_resource.data_type = cache_object.data_type
+            field_resource.id2value = cache_object.id2value
+            field_resource.hash = cache_object.hash
+            field_resource.processor_type = cache_object.processor_type
+            field_resource.has_text_resolver = cache_object.has_text_resolver
+            field_resource.has_embedding_resolver = cache_object.has_embedding_resolver
+            return field_resource
+
+        def to_metadata(self):
+            cache_object = Bunch(
+                index_name=self.index_name,
+                field_name=self.field_name,
+                data_type=self.data_type,
+                id2value=self.id2value,
+                hash=self.hash,
+                processor_type=self.processor_type,
+                has_text_resolver=self.has_text_resolver,
+                has_embedding_resolver=self.has_embedding_resolver
+            )
+            return cache_object
+
+        @staticmethod
+        def auto_string_processor(string_or_strings, query_type):
+
+            try:
+                english_stop_words = set(nltk_stopwords.words('english'))
```

Comment:
this function assumes english language, but this wouldnt suffice for spanish/japanese, no?

Reply:
Very good point. Even in Elasticsearch, we assume text is in English. I think it's good idea to mention this note, as we can look into extending support for non-english QA in future. I think it is better not to add this feature in this PR to just keeps things tractable!

Reply:
Currently, we use `english analyzer` in the ES version of the QuestionAnswerer for unstructured text which does a similar processing. If we want to support more languages, we'd have to look into this on the ES side as well.

Reply:
Yes. I added TODO tag in both question_answerer.py as well as in elasticsearch_helpers.py files about multilingual support. Will also add it as _Next Steps_ at the end of this PR's description.